core/src/test/scala/unit/kafka/controller/ControllerIntegrationTest.scala

```diff
@@ -1045,6 +1044,73 @@ class ControllerIntegrationTest extends QuorumTestHarness {
     assertEquals(expectedAlterPartitionResponse, future.get(10, TimeUnit.SECONDS))
   }
 
+  @Test
+  def testShutdownBrokerNotAddedToIsr(): Unit = {
```

Comment:
I wonder if we can just test this case in one of the variants in `testAlterPartitionErrors`

Reply:
Perhaps? Do you think the current way isn't readable?